health/health.go

```diff
@@ -84,3 +89,11 @@ func writeHealthy(w http.ResponseWriter) {
 type Checker interface {
 	CheckHealth() error
 }
+
+// CheckerFunc implements Checker by calling a function.
+type CheckerFunc func() error
```

Comment:
Should it take a context?

Reply:
The `Checker` interface intentionally does not take in a `Context` because a `Checker` should not perform anything expensive like an RPC: it should purely report on the health at a snapshot in time.

Reply:
SGTM.